src/main/jhipster/src/main/webapp/app/entities/program/program.component.ts

```diff
@@ -57,19 +61,31 @@ export class ProgramComponent implements OnInit {
         private languageService: JhiLanguageService,
         private router: Router,
         private route: ActivatedRoute,
-        public context: ProgramContext
+        public context: ProgramContext,
+        public programUsageService: ProgramUsageService,
     ) {
     }
 
     async ngOnInit() {
         // We get user last opened project ff programUUID is not present in the local storage
-        if (!localStorage['programUUID']) {
-            const identity = await this.principal.identity();
-            this.user = identity;
-        }
-
+        const identity = await this.principal.identity();
+        this.user = identity;
         this.crops = await this.cropService.getCrops().toPromise();
 
+        this.programUsageService.getLastestSelectedProgram(this.user.id).toPromise().then((response) => {
+            const userProgramSelected = response.body;
+            if (userProgramSelected) {
+                this.getPrograms(1, undefined).subscribe((res) => {
+                    if (this.programsById[userProgramSelected.uniqueID]) {
+                        this.programModel = userProgramSelected.uniqueID;
+                        this.context.program = userProgramSelected;
+                        this.initialData = [{ id: userProgramSelected.uniqueID, text: userProgramSelected.name }]
+                    }
+                }, (err) => {
+                });
+            }
+        });
+
```

Comment:
I think we can omit the call to getPrograms if set programsById directly
```typescript
        this.programUsageService.getLastestSelectedProgram(this.user.id).toPromise().then((response) => {
            const userProgramSelected = response.body;
            if (userProgramSelected) {
                this.programsById[userProgramSelected.uniqueID] = userProgramSelected;
                this.programModel = userProgramSelected.uniqueID;
                this.context.program = userProgramSelected;
                this.initialData = [{ id: userProgramSelected.uniqueID, text: userProgramSelected.name }]
              
            }
        });


```

Reply:
As we discussed, calling to `getPrograms` has these benefits:
* we validate the program stored in the user program info table match with the first program returned by the /programs api
* we validate that the user still has access to the said program